var/spack/repos/builtin/packages/armpl-gcc/package.py

```diff
@@ -177,28 +228,36 @@
 
 
 def get_os(ver):
-    spack_os = spack.platforms.host().default_os
+    platform = spack.platforms.host()
+    if platform.name == "darwin":
+        return "macOS"
     if ver.startswith("22."):
-        return _os_map_before_23.get(spack_os, "")
+        return _os_map_before_23.get(platform.default_os, "")
     else:
-        return _os_map.get(spack_os, "RHEL-7")
+        return _os_map.get(platform.default_os, "RHEL-7")
 
 
 def get_package_url(version):
     base_url = "https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/"
     armpl_version = version.split("_")[0]
     armpl_version_dashed = armpl_version.replace(".", "-")
-    gcc_version = version.split("_")[1]
+    compiler_version = version.split("_", 1)[1]
     os = get_os(armpl_version)
-    filename = "arm-performance-libraries_" + armpl_version + "_" + os + "_" + gcc_version + ".tar"
+    if os == "macOS":
+        if armpl_version.startswith("23.06"):
+            return f"{base_url}{armpl_version_dashed}/armpl_{armpl_version}_{compiler_version}.dmg"
+        else:
+            filename = f"arm-performance-libraries_{armpl_version}_macOS.dmg"
+            return f"{base_url}{armpl_version_dashed}/macos/{filename}"
+    filename = "arm-performance-libraries_{}_{}_{}.tar".format(armpl_version, os, compiler_version)
```

Comment:
Thanks for switching to f-strings.  Curious why this one still uses `format()`.  Perhaps I'm missing something?

Reply:
@tldahlgren There are a number of places that still use `format()` in this package. I can replace all those in another patch.